src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -407,6 +408,14 @@
                          (initialize-schema {:datasource db-pool} config)
                          true
                          (catch java.sql.SQLTransientConnectionException ex
+                           (when (->> ex
+                                      Throwable->map
+                                      :cause
+                                      (re-find #"Your PuppetDB database contains a schema migration numbered"))
+                             ;; TODO need to add rlb's tk error handling stuff to throw with msg/status here
+                             ;; and need to add a new error kind in the updated start-puppetdb func once it's merged
+                             (throw (java.lang.IllegalStateException.
```

Comment:
Once #3204 is merged this we will need to throw an ex-info here that's recognized by the new version of the `start-puppetdb` func.

Reply:
We want to rearrange this func to allow the changes rlb made to catch the error in the start up case and not add the connectionInitSql check on the startup pool.